### PR TITLE
Integration of PULP cluster with HMR + RedMulE

### PR DESCRIPTION
**Pull request by yvantor**

### CODE REVIEW

**Comment by alex96295** [comment]
This is the last large 'chunk' of features we need in carfield, so as soon as it is ready, we merge it. Let me know when

**Comment by yvantor** [comment]
> This is the last large 'chunk' of features we need in carfield, so as soon as it is ready, we merge it. Let me know when

Sure, I will merge it probably today if the CI with the latest updates concludes in time!

File: hw/carfield.sv
Comment (luca-valente) [inline hw/carfield.sv:1217]: Def not necessary, but do we want to connect it to the PLIC? @alex96295

File: hw/carfield.sv
Comment (luca-valente) [inline hw/carfield.sv:1218]: Do we want to connect it to a reg so that we can read it ?

File: hw/regs/carfield_regs.csv
Comment (luca-valente) [inline hw/regs/carfield_regs.csv:51]: I would put it at the end of the memory map, so that the `fetch` signals are grouped together

**Review by luca-valente — COMMENTED** [review]
Few comments from my side :smile:

File: hw/carfield.sv
Comment (alex96295) [reply hw/carfield.sv:1218]: the busy signal? We can do it, it should be a reg in the `soc_ctrl` register file

File: hw/carfield.sv
Comment (luca-valente) [reply hw/carfield.sv:1217]: Routing the mailbox interrupt to the plic is much more necessary indeed.

File: hw/regs/carfield_regs.csv
Comment (luca-valente) [reply hw/regs/carfield_regs.csv:51]: @yvantor :crab:

Comment (yvantor) [reply hw/regs/carfield_regs.csv:51]: Sure!

Comment (yvantor) [reply hw/regs/carfield_regs.csv:51]: > @yvantor 🦀

:golfing_man:

File: hw/carfield.sv
Comment (yvantor) [reply hw/carfield.sv:1218]: I agree on connecting both eoc and busy to the register file, doing it now.

Comment (yvantor) [reply hw/carfield.sv]: Done

File: hw/regs/carfield_regs.csv
Comment (yvantor) [reply hw/regs/carfield_regs.csv]: Done

**Comment by yvantor** [comment]
@luca-valente can you check and tell me if there are any other changes required? Otherwise I can resolve the conversations and merge

File: hw/carfield.sv
Comment (yvantor) [reply hw/carfield.sv]: I reslove this since it is addressed in another PR.

File: hw/carfield.sv
Comment (alex96295) [inline hw/carfield.sv:1219]: So we do not connect this to the interrupt controller?

**Review by alex96295 — APPROVED** [review]
Other that the small comment I made, lgtm from me

Check also manually that elaboration finishes successfully, but it should be already checked by the CI

File: hw/carfield.sv
Comment (yvantor) [reply hw/carfield.sv:1219]: The only signal that would make sense to connect to the plic is the eoc, the others you highlighted must be registers that we configure to boot the cluster. Also the busy, I think it makes sense to have it as a register that we might want to check, not as an irq. For now I wired the eoc to a register as well so that we can check it, we can always bind it to the plic but I can also do it before closing the PR, let me know!

Comment (alex96295) [reply hw/carfield.sv:1219]: It's okay like this, in case we revert this in PR #8 

lgtm, we can merge

**Comment by yvantor** [comment]
Closing this.

**Pull request merged**
